### PR TITLE
Support "Configuration" in "newstore" feature

### PR DESCRIPTION
**Pull request by WeiZhang555**
Store the configuration file in persist.go.

This is the last file we should support for "newstore" feature, after this, we can safely remove old persistence storage drivers.

### CODE REVIEW

File: virtcontainers/persist.go [hunk 1]
```diff
@@ -171,6 +172,86 @@ func (s *Sandbox) dumpNetwork(ss *persistapi.SandboxState) {
...
+	}
+
+	if sconfig.AgentType == "kata" {
+		var sagent KataAgentConfig
+		err := mapstructure.Decode(sconfig.AgentConfig, &sagent)
+		if err != nil {
+			s.Logger().WithError(err).Error("internal error: KataAgentConfig failed to decode")
+		} else {
+			ss.Config.KataAgentConfig = &persistapi.KataAgentConfig{
+				LongLiveConn: sagent.LongLiveConn,
```
Comment (jodh-intel) [inline virtcontainers/persist.go]: Anyone else think of Star Trek when they see this? :smile: :rocket:

```go
spock.liveLongAndProsper(conn)

File: virtcontainers/persist.go [hunk 2]
```diff
@@ -335,3 +417,85 @@ func (s *Sandbox) supportNewStore() bool {
...
+		InitrdPath:            hconf.InitrdPath,
+		SharedFS:              hconf.SharedFS,
+		VirtioFSDaemon:        hconf.VirtioFSDaemon,
+		DisableBlockDeviceUse: hconf.DisableBlockDeviceUse,
+		UseVSock:              hconf.UseVSock,
+		DisableVhostNet:       hconf.DisableVhostNet,
+	}
+
+	if savedConf.AgentType == "kata" {
+
```
Comment (jodh-intel) [inline virtcontainers/persist.go]: Nit: blank line.

File: cli/main.go [hunk 3]
```diff
@@ -401,6 +407,24 @@ func setupTracing(context *cli.Context, rootSpanName string) error {
...
+// add supported experimental features in context
+func addExpFeatures(clictx *cli.Context, runtimeConfig oci.RuntimeConfig) error {
+	ctx, err := cliContextToContext(clictx)
+	if err != nil {
+		return err
+	}
+
+	var exps []string
+	for _, e := range runtimeConfig.Experimental {
+		exps = append(exps, e.Name)
```
[the remark this answers is not in the document]
Comment (WeiZhang555) [reply cli/main.go]: Actually the virtcontainers package already has a debug log message for this 😄

File: virtcontainers/persist.go (hunk 1 above)
Comment (WeiZhang555) [reply virtcontainers/persist.go]: 🙌 :raised hand:

File: virtcontainers/persist.go (hunk 2 above)
Comment (WeiZhang555) [reply virtcontainers/persist.go]: Will remove in next push

**Comment by WeiZhang555** [comment]
@jodh-intel Thanks for help reviewing! There're some breaking cases, I will fix them in my free time.

I will also add some unit tests later.

**Comment by WeiZhang555** [comment]
image

After a hard debugging, all CI passed now (enabled experimental features for all Jenkins jobs).

I'll squash the commits and remove the WIP labels, then waiting for merging now. :-)

**Pull request merged**